From: Mike Hommey <mh@glandium.org>
Date: Mon, 16 Dec 2013 10:53:58 +0900
Subject: Add debian/extra-stuff to upstream build system directory traversal

---
 moz.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/moz.build b/moz.build
index 37e12e4..2c1f652 100644
--- a/moz.build
+++ b/moz.build
@@ -221,4 +221,7 @@ SPHINX_TREES["update-infrastructure"] = "docs/update-infrastructure"
 
 SPHINX_TREES["content-security"] = "docs/content-security"
 
+if not CONFIG["LIBXUL_SDK"]:
+    DIRS += ["debian/extra-stuff"]
+
 include("build/templates.mozbuild")
